runtime/metricsview/executor_summary.go

```diff
@@ -0,0 +1,250 @@
+package metricsview
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+	"github.com/rilldata/rill/runtime"
+	"github.com/rilldata/rill/runtime/drivers"
+)
+
+const (
+	// DefaultSummaryLimit is the maximum number of dimensions to summarize.
+	DefaultSummaryLimit = 15
+	// DefaultSampleTimeWindow is the time window used for sampling values
+	DefaultSampleTimeWindow = 24 * time.Hour
+	// statsPerDimension is the number of statistics returned per dimension
+	statsPerDimension = 4
+	// summaryQueryLimit is the maximum number of rows to return in the summary query
+	summaryQueryLimit = 1
+)
+
+// SummaryResult is the statistics for a single metrics view.
+type SummaryResult struct {
+	Dimensions []DimensionSummary `json:"dimensions"`
+	TimeRange  TimestampsResult   `json:"time_range"`
+}
+
+// DimensionSummary provides statistics for a single dimension in the metrics view.
+type DimensionSummary struct {
+	Name      string           `json:"name"`
+	DataType  string           `json:"data_type"`
+	Value     any              `json:"value,omitempty"`
+	HasNulls  bool             `json:"has_nulls,omitempty"`
+	MinValue  any              `json:"min_value,omitempty"`
+	MaxValue  any              `json:"max_value,omitempty"`
+	TimeRange TimestampsResult `json:"time_range,omitempty"`
+}
+
+// Summary provides statistics for all dimensions and measures in the metrics view.
+func (e *Executor) Summary(ctx context.Context, timeDimension string) (*SummaryResult, error) {
```

Comment:
I don't think it should accept a `timeDimension` here – we support multiple time dimensions, but the AI likely doesn't know which to use in advance. So the summary should give it info about every time dimension available.

I.e. it should return min/max for every dimension where `dim.DataType` is a timestamp type, as well as the default `mv.TimeDimension` (note that `mv.TimeDimension` may or may not be present in `mv.Dimensions` – yeah that ambiguity is a bit confusing).

As for the time range filter that is used to limit other data scans, I suggest doing that using `mv.TimeDimension` as it's most likely to be indexed.